static/src/javascripts/projects/commercial/modules/article-body-adverts-wide.js

```diff
@@ -0,0 +1,241 @@
+            minBelow: adSizes.mpu.height,
+            selectors: {
+                ' > h2': {minAbove: isMobile ? 100 : 0, minBelow: 250},
+                ' .ad-slot': {minAbove: 500, minBelow: 500},
+                ' > :not(p):not(h2):not(.ad-slot)': {minAbove: 35, minBelow: 400}
+            },
+            filter: function(slot) {
+                if (!prevSlot || Math.abs(slot.top - prevSlot.top) - adSizes.mpu.height >= rules.selectors[' .ad-slot'].minBelow) {
+                    prevSlot = slot;
+                    return true;
+                }
+                return false;
+            }
+        };
+
+        if (!isMerch && isWide) {
+            rules.minBelow = 100;
+            rules.selectors = {
+                ' .ad-slot': { minAbove: 500, minBelow: 500 }
+            };
+        }
+
+        return rules;
+    }
+
+    function getInlineMerchRules() {
+        var inlineMerchRules = getRules(true);
+        inlineMerchRules.minAbove = 300;
+        inlineMerchRules.selectors[' > h2'].minAbove = 100;
+        inlineMerchRules.selectors[' > :not(p):not(h2):not(.ad-slot)'].minAbove = 200;
+        return inlineMerchRules;
+    }
+
+    function getLongArticleRules() {
+        var longArticleRules = getRules();
+        longArticleRules.selectors[' .ad-slot'].minAbove =
+        longArticleRules.selectors[' .ad-slot'].minBelow = Math.max(500, detect.getViewport().height);
+        return longArticleRules;
+    }
+
+    // Add new ads while there is still space
+    function addArticleAds(count, rules) {
+        return spaceFiller.fillSpace(rules, insertInlineAds, {
+            waitForImages: true,
+            waitForLinks: true,
+            waitForInteractives: true
+        });
+
+        function insertInlineAds(paras) {
+            var slots = paras
+            .slice(0, Math.min(paras.length, count))
+            .map(function (para) {
+                return insertAdAtPara(para, getSlotName(), 'inline');
+            });
+
+            if (isWide) {
+                inlineSlots.push.apply(inlineSlots, slots);
+            }
+        }
+    }
+
+    function insertAdAtPara(para, name, type) {
+        var ad = createSlot(name, type);
+        para.parentNode.insertBefore(ad, para);
+        return ad;
+    }
+
+    function addSlots() {
+        qwery('.js-ad-slot', mainColumn).forEach(addSlot);
+    }
+
+    function addInlineMerchAd() {
+        return spaceFiller.fillSpace(getInlineMerchRules(), function (paras) {
+            return insertAdAtPara(paras[0], 'im', 'im');
+        }, {
+            waitForImages: true,
+            waitForLinks: true,
+            waitForInteractives: true
+        });
+    }
+
+    function addInlineAds() {
+        return addArticleAds(2, getRules())
+        .then(function () {
+            if (inlineSlots.length === 2) {
+                return addArticleAds(8, getLongArticleRules());
+            }
```

Comment:
Given that long article ads are more constrained, if less than 2 ad slots where added on the first call, then it is impossible that others will be added on the second.